Formula/r/redress.rb

```diff
@@ -0,0 +1,63 @@
+class Redress < Formula
+  desc "Tool for analyzing stripped Go binaries compiled with the Go compiler"
+  homepage "https://go-re.tk/redress/"
+  url "https://github.com/goretk/redress/archive/refs/tags/v1.1.1.tar.gz"
+  sha256 "5c938791f2010233054ede11401ed7a00e32151f37042f95e08d103e46d98f53"
+  license "AGPL-3.0-only"
+  head "https://github.com/goretk/redress.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "go" => [:build, :test]
+
+  def install
+    # https://github.com/goretk/redress/blob/develop/Makefile#L11-L14
+    redress_version = version
+    gore_version = File.read(buildpath/"go.mod").scan(%r{goretk/gore v(\S+)}).flatten.first
+    go_version = Formula["go"].version
+
+    ldflags = %W[
+      -s -w
+      -X main.redressVersion=#{redress_version}
+      -X main.goreVersion=#{gore_version}
+      -X main.compilerVersion=#{go_version}
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags)
+
+    generate_completions_from_executable(bin/"redress", "completion")
+  end
+
+  test do
+    assert_match "Version:  #{version}", shell_output("#{bin}/redress version")
+
+    if Hardware::CPU.arm?
+      # redress can't currently process ARM binaries, so we cross-compile this test binary
```

Comment:
Can it check `test_fixtures("elf/hello")`?

Reply:
Not unless that test fixture is a golang binary.

Reply:
Is there some hello-world binary we can download somewhere?

Reply:
This is in line with how tests are implemented for other Golang formula like this; which is what I based it off.

I'm not sure how it would be a better option to download an arbitrary binary just to run in the test?

Everything I've done here (including alternative options) was already discussed earlier on this PR and the approach taken was based on that discussion.

Reply:
it does not have to be that arbitrary (as it can be trusted domain hosted on github.com or something alike) and we also use checksum to guard the integrity of the binary.

Let me see what I can help with.

(Sorry about the hassle, we are trying to get it right from the beginning :) )

Reply:
> it does not have to be that arbitrary (as it can be trusted domain hosted on github.com or something alike) and we also use checksum to guard the integrity of the binary.

@chenrui333 I guess by 'arbitrary', I was less concerned about file integrity (because checksum/etc), and more that it just seems like a worse solution to download a random binary from somewhere that's unrelated to the project/homebrew, just to avoid using an existing pattern of compiling a small binary to be used in the tests.

> Let me see what I can help with.

@chenrui333 I'd appreciate that.

> (Sorry about the hassle, we are trying to get it right from the beginning :) )

@chenrui333 _nods_ even though it's frustrating from a "trying to contribute" perspective, I do understand the 'why' of it, at least at a high level (even if I don't personally agree with the specifics of it in this instance)

Reply:
See this comment for latest updates, which sort of sidesteps the core testing issue being discussed here:

- https://github.com/Homebrew/homebrew-core/pull/155724#issuecomment-1839844939